cmake/FindCython.cmake

```diff
@@ -5,7 +5,7 @@
 # This finds the "cython" executable in your PATH, and then in some standard
 # paths:
 SET(CYTHON_BIN cython CACHE STRING "Cython executable name")
-SET(CYTHON_FLAGS --cplus --fast-fail)
+SET(CYTHON_FLAGS --cplus --fast-fail -3)
```

Comment:
This only works with Cython 0.29beta1, or does the `-3` flag also work with previous versions? I.e., whether we can enable this by default, or whether we have to check the Cython version.

Reply:
cython-0.19.1 (minimum req. according to symengine.py's setup.py) does have the `-3` flag.

Reply:
Looks like we should use the `-2` flag for python2 builds and `-3` flag for python3 builds, but I haven't looked into that.